pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -0,0 +1,1089 @@
+	return fmt.Sprintf("%s-pravega-segmentstore", p.Name)
+}
+
+func (p *PravegaCluster) PravegaControllerServiceURL() string {
+	return fmt.Sprintf("tcp://%v.%v:%v", p.ServiceNameForController(), p.Namespace, "9090")
+}
+
+func (p *PravegaCluster) LabelsForController() map[string]string {
+	labels := p.LabelsForPravegaCluster()
+	labels["component"] = "pravega-controller"
+	return labels
+}
+
+func (p *PravegaCluster) LabelsForSegmentStore() map[string]string {
+	labels := p.LabelsForPravegaCluster()
+	labels["component"] = "pravega-segmentstore"
+	return labels
+}
+
+func (pravegaCluster *PravegaCluster) LabelsForPravegaCluster() map[string]string {
+	return map[string]string{
+		"app":             "pravega-cluster",
+		"pravega_cluster": pravegaCluster.Name,
+	}
+}
+
+func (p *PravegaCluster) PdbNameForController() string {
+	return fmt.Sprintf("%s-pravega-controller", p.Name)
+}
+
+func (p *PravegaCluster) ConfigMapNameForController() string {
+	return fmt.Sprintf("%s-pravega-controller", p.Name)
+}
+
+func (p *PravegaCluster) ServiceNameForController() string {
+	return fmt.Sprintf("%s-pravega-controller", p.Name)
+}
+
+func (p *PravegaCluster) ServiceNameForSegmentStore(index int32) string {
+	if util.IsVersionBelow07(p.Spec.Version) {
+		return p.ServiceNameForSegmentStoreBelow07(index)
+	}
+	return p.ServiceNameForSegmentStoreAbove07(index)
+}
+
+func (p *PravegaCluster) ServiceNameForSegmentStoreBelow07(index int32) string {
+	return fmt.Sprintf("%s-pravega-segmentstore-%d", p.Name, index)
+}
+
+func (p *PravegaCluster) ServiceNameForSegmentStoreAbove07(index int32) string {
+	return fmt.Sprintf("%s-pravega-segment-store-%d", p.Name, index)
+}
+
+func (p *PravegaCluster) HeadlessServiceNameForSegmentStore() string {
+	return fmt.Sprintf("%s-pravega-segmentstore-headless", p.Name)
+}
+
+func (p *PravegaCluster) HeadlessServiceNameForBookie() string {
+	return fmt.Sprintf("%s-bookie-headless", p.Name)
+}
+
+func (p *PravegaCluster) DeploymentNameForController() string {
+	return fmt.Sprintf("%s-pravega-controller", p.Name)
+}
+
+func (p *PravegaCluster) PdbNameForSegmentstore() string {
+	return fmt.Sprintf("%s-segmentstore", p.Name)
+}
+
+func (p *PravegaCluster) ConfigMapNameForSegmentstore() string {
+	return fmt.Sprintf("%s-pravega-segmentstore", p.Name)
+}
+
+func (p *PravegaCluster) GetClusterExpectedSize() (size int) {
+	return int(p.Spec.Pravega.ControllerReplicas + p.Spec.Pravega.SegmentStoreReplicas)
+}
+
+func (p *PravegaCluster) PravegaImage() (image string) {
+	return fmt.Sprintf("%s:%s", p.Spec.Pravega.Image.Repository, p.Spec.Version)
+}
+
+func (p *PravegaCluster) PravegaTargetImage() (string, error) {
+	if p.Status.TargetVersion == "" {
+		return "", fmt.Errorf("target version is not set")
+	}
+	return fmt.Sprintf("%s:%s", p.Spec.Pravega.Image.Repository, p.Status.TargetVersion), nil
+}
+
+func (p *PravegaCluster) NewEvent(name string, reason string, message string, eventType string) *corev1.Event {
```

Comment:
why is this function NewEvent() added?

Reply:
This is not a new function. Just moved existing function from util package to here.
This was required because of controller-runtime upgrade.